### ISSUE TITLE
BatchList - sort by Platform or Project examines `id`, not `name`

### ISSUE THREAD

**Issue by IanMayo** [DeepBlueCLtd/RCO#994]
In the `BatchList` page the user can choose to sort by `Platform` and `Project`.  But the items aren't displayed in alphabetical order.

I'm pretty sure they are being sorted by `id`, not `name`.

**Comment by TahaKhanAbdalli** [comment]
Hi @IanMayo, I have just gone through the issue and came to know that the soul is not sorting the foreign key.

Meaning, in our current case, we are getting data from other tables on the basis of `Id` and in reference to the `Id`, we show data on the batch list. So when we sort the project or platform or any other column which is a` foreign key` in current table, it sorts the `Id` first, then gets the data from other tables (referenced table) according to that `sorted Id`.

![image](https://github.com/DeepBlueCLtd/RCO/assets/50602678/d1de5c02-d4cf-4a0c-ae55-ee97c00f874a)

**Comment by IanMayo** [comment]
Yes, that matches my understanding.  When asking for data to be sorted by `User` the users are expecting to be sorted into `Andrew`, `Brian`, `Christine`, `Delia`, etc.  But, actually it's sorted by `1, 3, 5, 12` - which could get rendered as `Brian`, `Delia`, `Christine`, `Andrew`.

The only way I can think of solving this is to introduce more `RichItem` style SQL `VIEW`s - which includes the FK names - so as well as `user` we have `userName` - and we allow the users to sort by these `rich` values.

**Comment by TahaKhanAbdalli** [comment]
Yes, @IanMayo  we can do that.

**Comment by IanMayo** [comment]
Ok, @TahaKhanAbdalli - but let's defer this for a future phase please.

This and #997 both require a related fix `rich tables`.  But, I don't have customer endorsement for working on it yet.
